Refactor the canvas test generation code

gsnedders opened the pull request:
This unifies the two largely-identical implementations that generate the canvas (&lt;canvas> and OffscreenCanvas) tests, and gets rid of the duplication of templates.

This does, unfortunately, mean a small number of tests (~140 of the several thousand generated) have whitespace changes.

Hopefully this makes it easier for any future work on these scripts!

gsnedders committed:
Refactor the canvas test generation code

This unifies the two largely-identical implementations that generate the canvas
(<canvas> and OffscreenCanvas) tests, and gets rid of the duplication of
templates.

This does, unfortunately, mean a small number of tests (~140 of the several
thousand generated) have whitespace changes.

Hopefully this makes it easier for any future work on these scripts.

jgraham commented:
Would be useful to refactor this into one commit for the script changes and one commit for the generated output files.

gsnedders committed:
Regenerate canvas tests

This is entirely changes to blank lines (see `git-diff --ignore-blank-lines`),
except for 2d.path.roundrect.radius.noargument.html which appears to have never
been generated since 31eb342c6aa when the test name was modified.

gsnedders commented:
> Would be useful to refactor this into one commit for the script changes and one commit for the generated output files.

Done!

gsnedders committed:
Reformat these test generation scripts before refactoring

This is running `pyupgrade --py36-plus` & `black` until they reached a fixed
point.

gsnedders commented:
Apparently `html/canvas/tools` got substantially rewritten through 2023–4, and this never got reviewed, and as with all the changes, you probably are just as well off starting from scratch to actually land something comparable.
